//tools/lint:util_test may fail on macOS

```
FAIL: //tools/lint:util_test (see /private/var/tmp/_bazel_user/7c7ad393d9017d670ea6cd4ae3566d69/execroot/drake/bazel-out/darwin-opt/testlogs/tools/lint/util_test/test.log)
INFO: From Testing //tools/lint:util_test:
==================== Test output for //tools/lint:util_test:
F
======================================================================
FAIL: test_find (__main__.UtilTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/private/var/tmp/_bazel_user/7c7ad393d9017d670ea6cd4ae3566d69/execroot/drake/bazel-out/darwin-opt/bin/tools/lint/util_test.runfiles/drake/tools/lint/test/util_test.py", line 34, in test_find
    one_relpath)
AssertionError: third_party/.DS_Store

----------------------------------------------------------------------
Ran 1 test in 0.241s

FAILED (failures=1)
================================================================================
```
Output could be a little more helpful rather than just a filename also.

--- a/tools/lint/test/util_test.py
+++ b/tools/lint/test/util_test.py
@@ -31,7 +31,7 @@ class UtilTest(unittest.TestCase):
                 self.assertTrue(
                     one_relpath in THIRD_PARTY_SOURCES_ALLOWED_TO_BE_FOUND or
                     one_relpath.startswith("."),
-                    one_relpath)
+                    one_relpath + " has been mis-identified as a source file")
 
 
 # TODO(jwnimmer-tri) Omitting or mistyping these lines means that no tests get

--- a/tools/lint/util.py
+++ b/tools/lint/util.py
@@ -48,6 +48,8 @@ def find_all_sources():
         rel_dirpath = abs_dirpath[len(workspace_root) + 1:]
         # Take all files within the currently-walked directory.
         for one_filename in files:
+            if one_filename == ".DS_Store":
+                continue
             relpaths.append(os.path.join(rel_dirpath, one_filename))
         # Don't recurse into children of "third_party".
         if abs_dirpath.endswith("/third_party"):

